commons/src/main/scala/com/lynbrookrobotics/potassium/commons/position/PurePursuitTasks.scala

```diff
@@ -0,0 +1,190 @@
+package com.lynbrookrobotics.potassium.commons.position
+
+import com.lynbrookrobotics.potassium.Signal
+import com.lynbrookrobotics.potassium.commons.cartesianPosition.XYPosition
+import com.lynbrookrobotics.potassium.commons.drivetrain._
+import com.lynbrookrobotics.potassium.tasks.FiniteTask
+import com.lynbrookrobotics.potassium.units.{Point, Segment}
+import squants.{Dimensionless, Each}
+import squants.space.{Angle, Degrees, Feet, Length}
+
+trait PurePursuitTasks extends UnicycleCoreTasks {
+  import controllers._
+
+  val origin = new Point(
+    Feet(0),
+    Feet(0)
+  )
+
+  def compassToTrigonometric(angle: Angle): Angle = {
+    Degrees(90) - angle
+  }
+
+  def clamp(toClamp: Dimensionless, maxMagnitude: Dimensionless): Dimensionless = {
+    toClamp min maxMagnitude max -maxMagnitude
+  }
+
+  class GoToPoint(target: Point, tolerance: Length)
+                 (implicit drive: Drivetrain,
+                         properties: Signal[UnicycleProperties],
+                         hardware: UnicycleHardware) extends FiniteTask {
+    override def onStart(): Unit = {
+      val initialTurnPosition = hardware.turnPosition.get
+
+      val position = XYPosition(
+        hardware.turnPosition.map(_ - initialTurnPosition).map(compassToTrigonometric),
+        hardware.forwardPosition
+      )
+
+      val (forwardOutput, forwardError) = pointDistanceControl(
+        position,
+        Signal.constant(target).toPeriodic)
+
+      val (turnOutput, heading, lookAheadPoint) = purePursuitControllerTurn(
+        hardware.turnPosition.map(_ - initialTurnPosition),
+        position,
+        Signal.constant((Segment(origin, target), None)))
+
+      val unicycleOutput = forwardOutput.zip(turnOutput).zip(lookAheadPoint).zip(heading).map { o =>
+        val (((forward, turn), _), fdMultiplier) = o
+        UnicycleSignal(fdMultiplier * clamp(forward, Each(0.4)), turn)
+      }.withCheck { _ =>
+        forwardError.peek.get.foreach { e =>
+          if (e.abs <= tolerance) {
+            println("finished GoToPoint!")
+            finished()
+          }
+        }
+      }
+
+      drive.setController(toDriveSignal(unicycleOutput))
+    }
+
+    override def onEnd(): Unit = {
+      drive.resetToDefault()
+    }
+  }
+
+  class Follow2WayPoints(wayPoints: List[Point], tolerance: Length)
```

Comment:
Since this only takes two, can you make this a (Point, Point) to make it impossible to have more/less than 2 points?